kubeflow/argo/argo.libsonnet

```diff
@@ -0,0 +1,429 @@
+                    name: "ARGO_NAMESPACE",
+                    valueFrom: {
+                      fieldRef: {
+                        apiVersion: "v1",
+                        fieldPath: "metadata.namespace",
+                      },
+                    },
+                  },
+                  {
+                    name: "IN_CLUSTER",
+                    value: "true",
+                  },
+                ],
+                image: "argoproj/argoui:v2.0.0-beta1",
+                imagePullPolicy: "IfNotPresent",
+                name: "argo-ui",
+                resources: {},
+                terminationMessagePath: "/dev/termination-log",
+                terminationMessagePolicy: "File",
+              },
+            ],
+            dnsPolicy: "ClusterFirst",
+            restartPolicy: "Always",
+            schedulerName: "default-scheduler",
+            securityContext: {},
+            serviceAccount: "argo-ui",
+            serviceAccountName: "argo-ui",
+            terminationGracePeriodSeconds: 30,
+            readinessProbe: {
+              httpGet: {
+                path: "/",
+                port: 8001,
+              },
+            },
+          },
+        },
+      },
+    },  // deployUi
+
+    uiService: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: {
+          app: "argo-ui",
+        },
+        name: "argo-ui",
+        namespace: namespace,
+      },
+      spec: {
+        ports: [
+          {
+            port: 80,
+            targetPort: 8001,
+          },
+        ],
+        selector: {
+          app: "argo-ui",
+        },
+        sessionAffinity: "None",
+        type: "NodePort",
+      },
+    },
+
+    config: {
+      apiVersion: "v1",
+      data: {
+        config: @"executorImage: argoproj/argoexec:v2.0.0-beta1",
+      },
+      kind: "ConfigMap",
+      metadata: {
+        name: "workflow-controller-configmap",
+        namespace: namespace,
+      },
+    },
+
+    serviceAccount: {
+      apiVersion: "v1",
+      kind: "ServiceAccount",
+      metadata: {
+        name: "argo",
+        namespace: namespace,
+      },
+    },  // service account
+
+    // Keep in sync with https://github.com/argoproj/argo/blob/master/cmd/argo/commands/const.go#L20
+    // Permissions need to be cluster wide for the workflow controller to be able to process workflows
+    // in other namespaces. We could potentially use the ConfigMap of the workflow-controller to
+    // scope it to a particular namespace in which case we might be able to restrict the permissions
+    // to a particular namespace.
+    role: {
+      apiVersion: "rbac.authorization.k8s.io/v1beta1",
+      kind: "ClusterRole",
+      metadata: {
+        labels: {
+          app: "argo",
+        },
+        name: "argo",
+        namespace: namespace,
+      },
+      rules: [
+        {
+          apiGroups: [""],
+          resources: [
+            "pods",
+            "pods-exec",
```

Comment:
Same here. I've only ever seen this pods/exec

Reply:
Yes, the docs say to "use a slash to delimit the resource and subresource"
https://kubernetes.io/docs/admin/authorization/rbac/#referring-to-resources

Reply:
Thanks for catching this; it was a bad copy and paste error on my part.